pkg/compute/executor.go

```diff
@@ -313,6 +316,16 @@ func (e *BaseExecutor) Run(ctx context.Context, state store.LocalExecutionState)
 
 	result, err := e.Wait(ctx, state)
```

Comment:
I am not sure `Wait` produces an error when the execution is canceled, rather the `result` error field will be set to the error received from the container when it was killed (in the case for docker), containing a non-zero exit code and stuff. Unfortunately the result of `Wait` is executor dependent.